src/main/java/com/cleanroommc/groovyscript/compat/mods/actuallyadditions/Empowerer.java

```diff
@@ -0,0 +1,145 @@
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import de.ellpeck.actuallyadditions.api.ActuallyAdditionsAPI;
+import de.ellpeck.actuallyadditions.api.recipe.EmpowererRecipe;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.Ingredient;
+import org.jetbrains.annotations.Nullable;
+
+public class Empowerer extends VirtualizedRegistry<EmpowererRecipe> {
+
+    public Empowerer() {
+        super();
+    }
+
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    @Override
+    @GroovyBlacklist
+    public void onReload() {
+        removeScripted().forEach(ActuallyAdditionsAPI.EMPOWERER_RECIPES::remove);
+        ActuallyAdditionsAPI.EMPOWERER_RECIPES.addAll(restoreFromBackup());
+    }
+
+    public EmpowererRecipe add(Ingredient input, ItemStack output, Ingredient modifier1, Ingredient modifier2, Ingredient modifier3, Ingredient modifier4, int energyPerStand, int time, float[] particleColor) {
+        EmpowererRecipe recipe = new EmpowererRecipe(input, output, modifier1, modifier2, modifier3, modifier4, energyPerStand, time, particleColor);
+        add(recipe);
+        return recipe;
+    }
+
+    public void add(EmpowererRecipe recipe) {
+        if (recipe == null) return;
+        addScripted(recipe);
+        ActuallyAdditionsAPI.EMPOWERER_RECIPES.add(recipe);
+    }
+
+    public boolean remove(EmpowererRecipe recipe) {
+        if (recipe == null) return false;
+        addBackup(recipe);
+        ActuallyAdditionsAPI.EMPOWERER_RECIPES.remove(recipe);
+        return true;
+    }
+
+    public boolean removeByInput(IIngredient input) {
+        return ActuallyAdditionsAPI.EMPOWERER_RECIPES.removeIf(recipe -> {
+            boolean found = recipe.getInput().test(IngredientHelper.toItemStack(input));
+            if (found) {
+                addBackup(recipe);
+            }
+            return found;
+        });
+    }
+
+    public boolean removeByOutput(ItemStack output) {
+        return ActuallyAdditionsAPI.EMPOWERER_RECIPES.removeIf(recipe -> {
+            boolean matches = ItemStack.areItemStacksEqual(recipe.getOutput(), output);
+            if (matches) {
+                addBackup(recipe);
+            }
+            return matches;
+        });
+    }
+
+    public void removeAll() {
+        ActuallyAdditionsAPI.EMPOWERER_RECIPES.forEach(this::addBackup);
+        ActuallyAdditionsAPI.EMPOWERER_RECIPES.clear();
+    }
+
+    public SimpleObjectStream<EmpowererRecipe> streamRecipes() {
+        return new SimpleObjectStream<>(ActuallyAdditionsAPI.EMPOWERER_RECIPES)
+                .setRemover(this::remove);
+    }
+
+    public static class RecipeBuilder extends AbstractRecipeBuilder<EmpowererRecipe> {
+
+        private IIngredient mainInput;
+        private int energyPerStand;
+        private int time;
+        private float[] particleColor = new float[]{0, 0, 0};
```

Comment:
Is this really how AA stores their color? :concern:

Reply:
it is, yup. should i add builder methods to edit it in a more conventional manner (eg Color, hexadecimal, r/g/b)?

Reply:
I would store the color as an `int` value and construct the cursed format when you register the recipe.

Reply:
Yes, please add sane mathod params